src/main/java/com/oltpbenchmark/api/BenchmarkModule.java

```diff
@@ -73,15 +75,24 @@ public abstract class BenchmarkModule {
     public BenchmarkModule(WorkloadConfiguration workConf) {
         this.workConf = workConf;
         this.dialects = new StatementDialects(workConf);
+        if (workConf.getXmlConfig().getBoolean("use_hikari_pool", false)) {
```

Comment:
Since worker inherits from benchmarkmodule are we creating one hikari pool per worker?

Reply:
even though worker inherits from benchmarkmodule, the datasource is created in benchmarkmodule's constructor which is called only once. Hence, there is only one hikari pool.